libsinsp/version.h is fragile with respect to use of __STDC_FORMAT_MACROS

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks! -->

**Describe the bug**

version.h at commit c8af440082fd996ade25b2128167168db98710a5 is a bit fragile with respect to its use of __STDC_FORMAT_MACROS. If that preprocessor define is set before including intttypes.h, it allows adding formatting directives like PRIu32 to sprintf/printf/etc.

The problem is that __STDC_FORMAT_MACROS must be defined before the *first* include of inttypes.h. If another header includes version.h but also includes inttypes.h before version.h *without* __STDC_FORMAT_MACROS, the formatting directives won't be available.

I suppose you could fix all includes of inttypes.h in libs to ensure that __STDC_FORMAT_MACROS is defined, but I would think an easier solution would be to move the implementation of the methods into a version.c file and control the includes there.

**How to reproduce it**

I can't include the full build log, since this is a custom use of libs, but I do get this error:
```
XXX/userspace/libsinsp/version.h: In constructor 'sinsp_version::sinsp_version(const string&)':
XXX/userspace/libsinsp/version.h:39:45: error: expected ')' before 'PRIu32'
   m_valid = sscanf(version_str.c_str(), "%" PRIu32 ".%" PRIu32 ".%" PRIu32,
```

If I add __STDC_FORMAT_MACROS to the include of inttypes.h in userspace/libscap/plugin_info.h and userspace/libscap/uthash.h, I can build without this error.

**Expected behaviour**

<!-- A clear and concise description of what you expected to happen. -->

**Screenshots**


<!-- If applicable, add screenshots to help explain your problem. -->

**Environment**

<!-- Please complete the following info. -->

- Falco version:
<!-- Use "falco --version". -->
- System info:
<!-- Falco has a built-in support command you can use  "falco --support | jq .system_info" -->
- Cloud provider or hardware configuration:
- OS:
<!-- Eg., output of "cat /etc/os-release". -->
- Kernel:
<!-- Eg., output of "uname -a". -->
- Installation method:
<!-- Eg., Kubernetes, RPM, DEB, from source? -->

**Additional context**

<!-- Add any other context about the problem here. -->

--- a/cmake/modules/CompilerFlags.cmake
+++ b/cmake/modules/CompilerFlags.cmake
@@ -45,6 +45,8 @@ if(NOT MSVC)
 		add_definitions(-DHAS_CAPTURE)
 	endif()
 
+	add_definitions(-D__STDC_FORMAT_MACROS)
+
 else() # MSVC
 	set(MINIMAL_BUILD ON)
 

--- a/userspace/chisel/chisel_fields_info.cpp
+++ b/userspace/chisel/chisel_fields_info.cpp
@@ -18,7 +18,6 @@ limitations under the License.
 //
 // Various helper functions to render stuff on the screen
 //
-#define __STDC_FORMAT_MACROS
 #include <stdio.h>
 #include <iostream>
 #include <assert.h>

--- a/userspace/common/types.h
+++ b/userspace/common/types.h
@@ -20,7 +20,6 @@ limitations under the License.
 #define strcasecmp _stricmp
 #endif
 
-#define __STDC_FORMAT_MACROS
 #include <inttypes.h>
 #include <stdbool.h>
 

--- a/userspace/libscap/scap_fds.c
+++ b/userspace/libscap/scap_fds.c
@@ -35,7 +35,6 @@ limitations under the License.
 #include <arpa/inet.h>
 #include <errno.h>
 #else
-#define __STDC_FORMAT_MACROS
 #include <inttypes.h>
 #include <unistd.h>
 #include <sys/param.h>

--- a/userspace/libsinsp/event.cpp
+++ b/userspace/libsinsp/event.cpp
@@ -18,7 +18,6 @@ limitations under the License.
 #include "sinsp_errno.h"
 
 #ifndef _WIN32
-#define __STDC_FORMAT_MACROS
 #include <inttypes.h>
 #include <sys/socket.h>
 #include <algorithm>

--- a/userspace/libsinsp/fdinfo.cpp
+++ b/userspace/libsinsp/fdinfo.cpp
@@ -16,7 +16,6 @@ limitations under the License.
 */
 
 #ifndef _WIN32
-#define __STDC_FORMAT_MACROS
 #include <inttypes.h>
 #include <algorithm>
 #endif

--- a/userspace/libsinsp/plugin.cpp
+++ b/userspace/libsinsp/plugin.cpp
@@ -18,8 +18,6 @@ limitations under the License.
 #ifndef _WIN32
 #include <dlfcn.h>
 #endif
-// This makes inttypes.h define PRIu32 (ISO C99 plus older g++ versions)
-#define __STDC_FORMAT_MACROS
 #include <inttypes.h>
 #include <string.h>
 #include <vector>

--- a/userspace/libsinsp/sinsp.h
+++ b/userspace/libsinsp/sinsp.h
@@ -54,8 +54,6 @@ limitations under the License.
 #include "sinsp_public.h"
 #include "sinsp_exception.h"
 
-#define __STDC_FORMAT_MACROS
-
 #include <string>
 #include <map>
 #include <queue>

--- a/userspace/libsinsp/threadinfo.cpp
+++ b/userspace/libsinsp/threadinfo.cpp
@@ -16,7 +16,6 @@ limitations under the License.
 */
 
 #ifndef _WIN32
-#define __STDC_FORMAT_MACROS
 #include <inttypes.h>
 #include <unistd.h>
 #endif

--- a/userspace/libsinsp/version.h
+++ b/userspace/libsinsp/version.h
@@ -21,7 +21,6 @@ limitations under the License.
 #include <string>
 #include <unordered_map>
 #include <cstdio>
-#define __STDC_FORMAT_MACROS
 #include <inttypes.h>
 
 /*!

